File: src/main/java/com/example/ajaxonboardingproject/model/Movie.java
package com.example.ajaxonboardingproject.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@Table(name = "movies")
public class Movie {
    @Id
    @EqualsAndHashCode.Exclude
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String description;
}

File: src/main/java/com/example/ajaxonboardingproject/model/Movie.kt
package com.example.ajaxonboardingproject.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document("movies")
data class Movie(
    var title: String,
    var description: String,
) {
    @Id
    lateinit var id: String
}
